protocol/messenger.go

```diff
@@ -719,11 +719,16 @@ func (m *Messenger) attachChatIdentity(cca *protobuf.ContactCodeAdvertisement) e
 	if err != nil {
 		return err
 	}
+
+	var hash []byte
 	if img == nil {
-		return errors.New("could not find image")
+		// Account has no images, so we use the identicon hash
+		hash = crypto.Keccak256([]byte(m.account.Identicon))
+	} else {
+		hash = img.Hash()
 	}
 
-	err = m.persistence.SaveWhenChatIdentityLastPublished(contactCodeTopic, img.Hash())
+	err = m.persistence.SaveWhenChatIdentityLastPublished(contactCodeTopic, hash)
 	if err != nil {
 		return err
 	}
```

Comment:
This seems like a bit of an antipattern. If we have no image we should probably save no hash rather than the hash of the identicon

Maybe something like below? What do you think?
```go
	if img == nil {
		return m.persistence.SaveWhenChatIdentityLastPublished(contactCodeTopic, []byte{})
	} else {
		return m.persistence.SaveWhenChatIdentityLastPublished(contactCodeTopic, img.Hash())
	}
```